app/controllers/creators_controller.rb

```diff
@@ -0,0 +1,42 @@
+class CreatorsController < ApplicationController
+  include HTTParty
+  LIMIT_PER_PAGE = 10
+
+  #use private methods to determine which api to search
+  def search
+    query = params[:query].gsub(/\W/, "")
+    if query == ""
+      flash["error"] = "Invalid search term."
+      redirect_to root_path
+    elsif params[:provider] == "vimeo"
+      @creators = vimeo_search(query).paginate(:page => params[:page], :per_page => 10)
+      render :search
+    else
+      @creators = twitter_search(query).paginate(:page => params[:page], :per_page => 10)
```

Comment:
I like to use the newer `page: params[:page]` syntax rather than the older `:page => params[:page]` syntax, but feel free to stick with what you do here if you prefer.